pkg/manager/member/pd_member_manager.go

```diff
@@ -138,11 +138,13 @@ func (pmm *pdMemberManager) syncPDServiceForTidbCluster(tc *v1alpha1.TidbCluster
 		svc := *oldSvc
 		svc.Spec = newSvc.Spec
 		// TODO add unit test
-		svc.Spec.ClusterIP = oldSvc.Spec.ClusterIP
 		err = controller.SetServiceLastAppliedConfigAnnotation(&svc)
 		if err != nil {
 			return err
 		}
+		if newSvc.Spec.ClusterIP == "" {
```

Comment:
I think it makes no sense for users to update `clusterIP`.

Reply:
If user fulfill the ClusterIP in the tidbcluster spec, shall we update the Service ClusterIP as user provided?

Reply:
ClusterIP cannot be updated after the initial set.
If users set ClusterIP during the cluster creation, the clusterIP will be set in the `CreateService` function.